src/mscorlib/shared/System/Memory.cs

```diff
@@ -100,6 +100,11 @@ internal Memory(OwnedMemory<T> owner, int index, int length)
             _length = length;
         }
 
+        // Gives Span.NonGeneric access to raw constructor. We don't want to expose the raw constructor itself as there are already multiple public constructors
+        // with the (X, int, int) signature. Exposing this overload would effectively shut down compile-time type-checking on those constructors.
```

Comment:
Is this why we are not changing `private Memory(object obj, int index, int length)` method to internal and instead adding this overload?

Reply:
Right - we have multiple `(<something>, int, int)` constructors - an `(object, int, int)` overload would be a very annoying attractive nuisance for the rest of CoreLib.

Reply:
IMHO, internal constructor that takes array and start would look better.

Reply:
Makes sense. Not sure now why the `(T[], int)` constructor was removed from `Memory` - the reason it was removed from `Span` don't seem to apply to `Memory`...

Reply:
I think that the Span should have the `(T[], int)` constructor as well - to implement the matching AsSpan methods.

Reply:
The `Span` constructor was removed on purpose (https://github.com/dotnet/corefx/issues/23471) - do we want to accept the "confusion tax" for internal purpose? I don't have a strong preference either way but the situation is slightly different from the `Memory` constructor.

Reply:
ok